src/mapping/VolumeCellInterpolation.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <list>
+#include <string>
+#include <vector>
+#include "logging/Logger.hpp"
+#include "mapping/BarycentricBaseMapping.hpp"
+#include "mapping/Polation.hpp"
+
+namespace precice {
+namespace mapping {
+
+/**
+ * @brief Mapping using orthogonal projection to nearest triangle/edge/vertex and
+ *        linear interpolation from projected point.
+ */
+class VolumeCellInterpolation : public BarycentricBaseMapping {
+public:
+  /// Constructor, taking mapping constraint.
+  VolumeCellInterpolation(Constraint constraint, int dimensions);
+
+  /// Destructor, empty.
+  virtual ~VolumeCellInterpolation() {}
```

Comment:
```suggestion
```
In case you don't want to further derive classes from this class, you can omit the `virtual`. Considering that the destructor is empty, you can just omit it as well.